spec/jobs/chat_channels/indexing_job_spec.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe ChatChannels::IndexingJob, type: :job do
+  include_examples "#enqueues_job", "chat_channels_indexing", chat_channel_id: 1
+end
```

Comment:
Won't we test algolia gem instead of testing our code doing this? In this case, we could test that the object receives `index!` call.

Reply:
@lightalloy not by cheating, since the build a new object inside the job. Yes, we would test that the algolia gem sends a specific HTTP call to webmock, but we also don't abstract algolia in the code, so I don't see it as a big problem right now

Reply:
When testing that `algolia` gem sends certain requests we rely on and test the `algolia` gem internal logic, which should be tested by the gem tests, not our app.
Testing that the request goes to algolia could be acceptable, so I tried testing it a bit and found out that algolia sends requests like this:
```
 GET https://5trk3h4z8c-dsn.algolia.net/1/indexes/SecuredChatChannel_test/settings?getVersion=1 with headers {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip,deflate', 'Content-Type'=>'application/json; charset=utf-8', 'Date'=>'Thu, 10 Oct 2019 10:22:52 GMT', 'User-Agent'=>'Algolia for Ruby (1.26.0); Ruby (2.6.3); Algolia for Rails (1.23.2); Rails (Rails::VERSION)', 'X-Algolia-Api-Key'=>'xxxxxxxxxxxxx', 'X-Algolia-Application-Id'=>'xxxxxx'} was made 1 time
```
In my opinion, testing such request would test a lot of internal details which we shouldn't test.

Reply:
I agree that it would be nice to have some tests for `#perfom_now` method.
One solution could be having a test like this:
```
  it "calls index" do
    channel = double
    allow(ChatChannel).to receive(:find_by).with(id: 1).and_return(channel)
    allow(channel).to receive(:index!)

    described_class.perform_now(chat_channel_id: 1)
    expect(channel).to have_received(:index!)
  end
```
I'm not a huge fan of such tests but at least it checks part of the job for the typos.

Reply:
A similar test could be added to test that the job doesn't fail when no `ChatChannel` was found.

Reply:
I agree with you @lightalloy.

@araslanov-e can you take a look at the suggestions up here?

We should at least have a mock test and one that tests the failure scenario

Thank you!